Source/Microsoft.Teams.Apps.CompanyCommunicator.Prep.Func/PreparingToSend/GetRecipientDataBatches/GetRecipientDataListForRosterActivity.cs

```diff
@@ -0,0 +1,169 @@
+        private readonly NotificationDataRepository notificationDataRepository;
+        private readonly SentNotificationDataRepository sentNotificationDataRepository;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GetRecipientDataListForRosterActivity"/> class.
+        /// </summary>
+        /// <param name="commonBotAdapter">The common bot adapter.</param>
+        /// <param name="botOptions">The bot options.</param>
+        /// <param name="notificationDataRepository">Notification data repository.</param>
+        /// <param name="sentNotificationDataRepository">Sent notification data repository.</param>
+        public GetRecipientDataListForRosterActivity(
+            CommonBotAdapter commonBotAdapter,
+            IOptions<BotOptions> botOptions,
+            NotificationDataRepository notificationDataRepository,
+            SentNotificationDataRepository sentNotificationDataRepository)
+        {
+            this.commonBotAdapter = commonBotAdapter;
+            this.microsoftAppId = botOptions.Value.MicrosoftAppId;
+            this.notificationDataRepository = notificationDataRepository;
+            this.sentNotificationDataRepository = sentNotificationDataRepository;
+        }
+
+        /// <summary>
+        /// Run the activity.
+        /// It uses Fan-out / Fan-in pattern to get recipient data list (team rosters) in parallel.
+        /// </summary>
+        /// <param name="context">Durable orchestration context.</param>
+        /// <param name="notificationDataEntityId">Notification data entity id.</param>
+        /// <param name="teamDataEntity">Team data entity.</param>
+        /// <param name="log">Logging service.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task RunAsync(
+            DurableOrchestrationContext context,
+            string notificationDataEntityId,
+            TeamDataEntity teamDataEntity,
+            ILogger log)
+        {
+            try
+            {
+                await context.CallActivityWithRetryAsync<IEnumerable<UserDataEntity>>(
+                    nameof(GetRecipientDataListForRosterActivity.GetRecipientDataListForRosterAsync),
+                    new RetryOptions(TimeSpan.FromSeconds(5), 3),
+                    new GetRecipientDataListForRosterActivityDTO
+                    {
+                        NotificationDataEntityId = notificationDataEntityId,
+                        TeamDataEntity = teamDataEntity,
+                    });
+            }
+            catch (Exception ex)
+            {
+                var stringBuilder = new StringBuilder();
+                stringBuilder.AppendLine($"Failed to load roster for team {teamDataEntity.TeamId}.");
+                stringBuilder.AppendLine(ex.Message);
+                var errorMessage = stringBuilder.ToString();
+
+                log.LogError(errorMessage);
```

Comment:
This should just be:
`log.LogError(ex, $"Failed to load roster for team {teamDataEntity.TeamId}: {ex.Message}");`